tests/file_tests/main.cpp

```diff
@@ -31,11 +34,25 @@ int main() {
         files++;
 
         trees.emplace_back(std::move(tree));
+        table.addSyntaxTree(&tree);
```

Comment:
Be careful here; you're adding a pointer to a temporary variable. It's only luck that it's working.